osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/MeterReadingMapping.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.kafka.da.application.mapping;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.opensmartgridplatform.adapter.kafka.IntervalBlock;
+import org.opensmartgridplatform.adapter.kafka.IntervalReading;
+import org.opensmartgridplatform.adapter.kafka.MeterReading;
+import org.opensmartgridplatform.adapter.kafka.ReadingType;
+import org.opensmartgridplatform.adapter.kafka.UsagePoint;
+import org.opensmartgridplatform.adapter.kafka.ValuesInterval;
+import org.opensmartgridplatform.domain.da.measurements.Measurement;
+import org.opensmartgridplatform.domain.da.measurements.MeasurementElement;
+import org.opensmartgridplatform.domain.da.measurements.MeasurementGroup;
+import org.opensmartgridplatform.domain.da.measurements.MeasurementReport;
+import org.opensmartgridplatform.domain.da.measurements.elements.FloatMeasurementElement;
+import org.opensmartgridplatform.domain.da.measurements.elements.TimestampMeasurementElement;
+
+import ma.glasnost.orika.CustomConverter;
+import ma.glasnost.orika.MappingContext;
+import ma.glasnost.orika.metadata.Type;
+
+public class MeterReadingMapping extends CustomConverter<MeasurementReport, MeterReading> {
+
+    @Override
+    public MeterReading convert(final MeasurementReport source, final Type<? extends MeterReading> destinationType,
+            final MappingContext mappingContext) {
+
+        final List<MeasurementElement> measurementElements = source.getMeasurementGroups()
+                .stream()
+                .map(MeasurementGroup::getMeasurements)
+                .flatMap(List::stream)
+                .map(Measurement::getMeasurementElements)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+
+        final List<IntervalReading> intervalReadings = Arrays.asList(this.getIntervalReading(measurementElements));
+
+        final String identification = source.getMeasurementGroups()
+                .get(0)
+                .getIdentification();
+        final ReadingType readingType = this.getReadingType(identification);
+        final IntervalBlock intervalBlock = new IntervalBlock(readingType, intervalReadings);
+
+        final Long start = intervalReadings.stream()
+                .map(IntervalReading::getTimeStamp)
+                .min(Long::compareTo)
+                .orElse(dateStringToEpoch("2020-01-13 00:00:00"));
+        final Long end = intervalReadings.stream()
+                .map(IntervalReading::getTimeStamp)
+                .max(Long::compareTo)
+                .orElse(dateStringToEpoch("2020-01-16 00:00:00"));
```

Comment:
seems a leftover of the POC

Reply:
Replaced with start datetime of next year